File: src/main/java/net/igneo/icv/entity/abyssStone/AbyssStoneModel.java
package net.igneo.icv.entity.abyssStone;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class AbyssStoneModel extends GeoModel<AbyssStoneEntity> {
  private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json");
  private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png");
  private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json");
  @Override
  public ResourceLocation getModelResource(AbyssStoneEntity abyssStoneEntity) {
    return model;
  }
  
  @Override
  public ResourceLocation getTextureResource(AbyssStoneEntity abyssStoneEntity) {
    return texture;
  }
  
  @Override
  public ResourceLocation getAnimationResource(AbyssStoneEntity abyssStoneEntity) {
    return animations;
  }
}

File: src/main/java/net/igneo/icv/entity/chestplate/abyssStone/AbyssStoneModel.kt
package net.igneo.icv.entity.chestplate.abyssStone

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class AbyssStoneModel : GeoModel<AbyssStoneEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json")

    override fun getModelResource(abyssStoneEntity: AbyssStoneEntity?): ResourceLocation {
        return model
    }

    override fun getTextureResource(abyssStoneEntity: AbyssStoneEntity?): ResourceLocation {
        return texture
    }

    override fun getAnimationResource(abyssStoneEntity: AbyssStoneEntity?): ResourceLocation {
        return animations
    }
}
